api/src/main/java/org/pmiops/workbench/api/DisksController.java

```diff
@@ -0,0 +1,143 @@
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace, String diskName) {
+    String googleProject =
+        workspaceService.lookupWorkspaceByNamespace(workspaceNamespace).getGoogleProject();
+    Disk disk =
+        leonardoMapper.toApGetDiskResponse(
+            leonardoNotebooksClient.getPersistentDisk(googleProject, diskName));
+
+    if (DiskStatus.FAILED.equals(disk.getStatus())) {
+      log.warning(
+          String.format("Observed failed PD %s in workspace %s", diskName, workspaceNamespace));
+    }
+
+    return ResponseEntity.ok(disk);
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> deleteDisk(String workspaceNamespace, String diskName) {
+    String googleProject =
+        workspaceService.lookupWorkspaceByNamespace(workspaceNamespace).getGoogleProject();
+    leonardoNotebooksClient.deletePersistentDisk(googleProject, diskName);
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> updateDisk(
+      String workspaceNamespace, String diskName, Integer diskSize) {
+    String googleProject =
+        workspaceService.lookupWorkspaceByNamespace(workspaceNamespace).getGoogleProject();
+    leonardoNotebooksClient.updatePersistentDisk(googleProject, diskName, diskSize);
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  @Override
+  public ResponseEntity<ListDisksResponse> listDisksInWorkspace(String workspaceNamespace) {
+    String googleProject =
+        workspaceService.lookupWorkspaceByNamespace(workspaceNamespace).getGoogleProject();
+
+    List<LeonardoListPersistentDiskResponse> responseList =
+        leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false);
+
+    List<Disk> diskList =
+        findTheMostRecentActiveDisks(
+            responseList.stream()
+                .map(leonardoMapper::toApiListDisksResponse)
+                .collect(Collectors.toList()));
+    ListDisksResponse listDisksResponse = new ListDisksResponse();
+    listDisksResponse.addAll(diskList);
+
+    return ResponseEntity.ok(listDisksResponse);
+  }
+
+  /**
+   * Finds the most recent disks for all apps and GCE runtime.
+   *
+   * <p>We use {@link Disk#getCreatedDate} as the most recent disk.
+   */
+  private List<Disk> findTheMostRecentActiveDisks(List<Disk> disksToValidate) {
+    // Iterate original list first to check if disks are valid. Print log if disks maybe in
+    // incorrect state to help future debugging.
+    // Disk maybe in incorrect state if having additional active state disks.
+    List<Disk> activateDisks =
+        disksToValidate.stream()
+            .filter(d -> ACTIVE_DISK_STATUS.contains(d.getStatus()))
+            .collect(Collectors.toList());
+    if (activateDisks.size() > (AppType.values().length + 1)) {
+      String diskNameList =
+          activateDisks.stream().map(Disk::getName).collect(Collectors.joining(","));
+      log.warning(String.format("Maybe incorrect disks: %s", diskNameList));
```

Comment:
It should never happen if both Leo and RW bug free. But because we are using random names for disks, that may hide potential bug. (so I think adding this warning message might be helpful).